src/libraries/Common/src/System/IO/Compression/ZLibNative.cs

```diff
@@ -320,17 +376,52 @@ public unsafe ErrorCode InflateEnd()
             public string GetErrorMessage() => _zStream.msg != ZNullPtr ? Marshal.PtrToStringUTF8(_zStream.msg)! : string.Empty;
         }
 
-        public static ErrorCode CreateZLibStreamForDeflate(out ZLibStreamHandle zLibStreamHandle, CompressionLevel level,
-            int windowBits, int memLevel, CompressionStrategy strategy)
+        public sealed class ZLibNativeException : Exception
```

Comment:
Why is this needed, rather than just throwing the ZLibException that should propagate?

Reply:
If only the class visibility is the problem, we were able to workaround that in the past by using SkipUseReferenceAssembly="true" on the ProjectReference tag in the csproj file.

Reply:
It's purely a problem with class visibility, but the [csproj](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.WebSockets/src/System.Net.WebSockets.csproj) doesn't have an explicit reference to System.IO.Compression, it just source-includes the files it needs. With that said, S.IO.C's part of the runtime, so I think it's safe to add a ProjectReference without having any knock-on impacts on the consumers of System.Net.WebSockets.

Adding the ProjectReference builds locally and lets me reference ZLibException - do I need to do anything else as a result of adding that dependency?

Reply:
I can't think of any possible problems at this moment, but please add a comment to the in the csproj file as to why we are bypassing the ref assembly.

> do I need to do anything else as a result of adding that dependency?

no, just the project reference should be enough.